### DISCUSSION TITLE
[Feature] Advanced Photo Grouping

### DISCUSSION THREAD

**Discussion by Peda1996** [immich-app/immich#6198] [Feature Request]
### The feature

This feature request proposes the integration of an advanced photo grouping system in Immich. The goal is to enable users to effortlessly organize and search through their photo collections based on timestamp, location, and visual similarity. This feature will enhance the user experience by providing more intuitive and efficient ways to manage large volumes of photos.

Should be in general pretty similar to how Samsung Gallerys app handles multiple similar images. Eg. it picks a "best" photo from the group and hide the other photos.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by bo0tzz** [comment]
This request is pretty generic, can you clarify what you mean by "advanced photo grouping system" and how that would look and work?

**Comment by Peda1996** [comment]
It would work by using eg the timestamps in combination with the
geolocation and an calculated hash that represents the photo.

If a couple of photos are made eg within a certain time this grouping
feature becomes active. Only one image is then visible in the gallery with
a small icon indicating the amount of subimages in the group. (see attached
image from the Samsung gallery).

When the image is then pressed all images are then shown.



bo0tzz ***@***.***> schrieb am Fr., 5. Jän. 2024, 17:17:

**Comment by zackpollard** [comment]
I already have plans to work on basically exactly this, however it'll come in stages. I am currently working on automatic grouping of burst photos, then will probably be working on the visually similar images, and then both of those will help towards this feature.
